packages/hw-app-eth/src/erc20.js

```diff
@@ -0,0 +1,80 @@
+// @flow
+import blob from "../data/erc20.js";
+
+/**
+ * Retrieve the token information by a given contract address if any
+ */
+export const byContractAddress = (contract: string): ?TokenInfo =>
+  get().byContract(asContractAddress(contract));
+
+/**
+ * list all the ERC20 tokens informations
+ */
+export const list = (): TokenInfo[] => get().list();
+
+export type TokenInfo = {
+  contractAddress: string,
+  ticker: string,
+  decimals: number,
+  chainId: number,
+  signature: Buffer,
+  data: Buffer
+};
+
+export type API = {
+  byContract: string => ?TokenInfo,
+  list: () => TokenInfo[]
+};
+
+const asContractAddress = (addr: string) => {
+  const a = addr.toLowerCase();
```

Comment:
this case thing has been an issue and discussed on Vault: should the front must handle this conversion, assuming the contract address has wrong case, or do we trust our unique source (which is crypto-assets-list)

we took the decision to never touch the contract address.
maybe it's not the exact same problem here, but I just put a warning on opening the door of manipulating the address.

Reply:
we need to keep it here because this lib is to give to the world.
e.g. MEW want to integrate the support of that, maybe they have a uppercase addr? we don't know, we need to converge to the same format. same for the 0x vs not having 0x.